python/tank/util/user_settings.py

```diff
@@ -55,7 +55,32 @@ def shotgun_proxy(self):
         """
         :returns: The default proxy.
         """
-        return self._get_value("http_proxy")
+
+        proxy = self._get_value("http_proxy")
+
+        if not proxy:
+            # Since the configuration settings do not specify an http proxy string,
+            # try to retrieve the operating system http proxy.
+
+            import urllib
+
+            # Get the dictionary of scheme to proxy server URL mappings; for example:
+            #     {"http": "http://foo:bar@74.50.63.111:80", "https": "http://74.50.63.111:443"}
+            # "getproxies" scans the environment for variables named <scheme>_proxy, in case insensitive way.
+            # When it cannot find it, for Mac OS X it looks for proxy information from Mac OSX System Configuration,
+            # and for Windows it looks for proxy information from Windows Systems Registry.
+            # If both lowercase and uppercase environment variables exist (and disagree), lowercase is preferred.
+            # Note the following restriction: "getproxies" does not support the use of proxies which
+            # require authentication (user and password) when looking for proxy information from
+            # Mac OSX System Configuration or Windows Systems Registry.
+            system_proxies = urllib.getproxies()
+
+            # Get the http proxy when it exists in the dictionary.
+            proxy = system_proxies.get("http")
+
+            logger.debug("Operating system proxy: %s" % self._get_filtered_proxy(proxy or "<not defined>"))
```

Comment:
this message will be logged every time someone requests the proxy - that doesn't seem ideal. Maybe it can log it once at startup. Is the reason that we do this system-proxy-lookup at runtime rather than at init that we want to be able to instantly pick up changes in the system? would be good to ephasize that in a comment!